add support for multiple munin usernames and passwords

delfuego opened the pull request:
This is a pull request to enable provisioning munin with multiple different username/password pairs for authentication.

It also uses openssl to generate the password hashes, eliminating the dependency on apache2-utils for its htpasswd utility.

delfuego committed:
add support for multiple munin usernames and passwords
use openssl to generate htpasswd file
remove apache2-utils dependency
update readme regarding username/password support

Soulou commented:
The command example seems broken, `-e HTTP_USERS=user1 user2` can't work, can you fix it ?

Being retrocompatible would be a real plus to avoid breaking any existing deployment

delfuego committed:
fix readme to reflect need to quote command-line, space-separated username/password env vars
add backward compatibility for previous MUNIN_USER and MUNIN_PASSWORD env vars

delfuego commented:
@Soulou Fixed both issues —  I added backward compatibility with the prior MUNIN_USER and MUNIN_PASSWORD variables, and the documentation is now updated to reflect the need to quote the command-line environment vars for the Munin username and password. (I've been deploying using an environment variable file, which does *not* require the variables to be quoted, hence the miss on that one.)

Soulou commented:
Thanks a lot, seems good to me
